crates/nu-parser/src/parse_keywords.rs

```diff
@@ -515,6 +526,19 @@ pub fn parse_def(
     let mut result = None;
 
     if let (Some(mut signature), Some(block_id)) = (sig.as_signature(), block.as_block()) {
+        for arg_name in &signature.required_positional {
+            verify_not_keyword(working_set, &arg_name.name, sig.span);
```

Comment:
This may be a future improvement but could we get a more specific span pointing to the part in the signature that is violating the rule?

Reply:
Not without rewriting `.as_signature` and `nu-protocol::signature`.  Currently the entire module has no notion of signatures.